NFe.Servicos/ServicosNFe.cs

```diff
@@ -490,6 +491,21 @@ private RetornoRecepcaoEvento RecepcaoEvento(int idlote, List<evento> eventos, S
             if (_cFgServico.ValidarSchemas)
                 Validador.Valida(servicoEvento, _cFgServico.VersaoRecepcaoEventoCceCancelamento, xmlEvento, cfgServico: _cFgServico);
 
+            var deveValidarDetEvento = DeveValidarDetalhamentoDoEvento(servicoEvento);
+
+            if (deveValidarDetEvento && _cFgServico.ValidarSchemas)
+            {
+                foreach (var evento in pedEvento.evento)
+                {
+                    var detEvento = evento.infEvento.detEvento;
+                    var detEventoXml = FuncoesXml.ClasseParaXmlString(detEvento);
+                    if (_cFgServico.RemoverAcentos)
+                        detEventoXml = detEventoXml.RemoverAcentos();
+
+                    Validador.Valida(servicoEvento, _cFgServico.VersaoRecepcaoEventoCceCancelamento, detEventoXml, cfgServico: _cFgServico, validarLote: false);
+                }
+            }
+
```

Comment:
A verificação se deve validar schemas pode ser simplificada, evita que verifique duas vezes.

Sugestão:
```suggestion
            if (_cFgServico.ValidarSchemas)
            {
                Validador.Valida(servicoEvento, _cFgServico.VersaoRecepcaoEventoCceCancelamento, xmlEvento, cfgServico: _cFgServico);
                
                var deveValidarDetEvento = DeveValidarDetalhamentoDoEvento(servicoEvento);

                if (deveValidarDetEvento)
                {
                    foreach (var evento in pedEvento.evento)
                    {
                        var detEvento = evento.infEvento.detEvento;
                        var detEventoXml = FuncoesXml.ClasseParaXmlString(detEvento);
                        if (_cFgServico.RemoverAcentos)
                            detEventoXml = detEventoXml.RemoverAcentos();

                        Validador.Valida(servicoEvento, _cFgServico.VersaoRecepcaoEventoCceCancelamento, detEventoXml, cfgServico: _cFgServico, validarLote: false);
                    }
                }
            }
```

Reply:
Ajustado no commit 436f9c5c